src/cloud.c

```diff
@@ -175,25 +182,13 @@ int cloud_set_cbs(cloud_downstream_cb_t on_update,
 		return -1;
 	}
 
-	err = amqp_set_queue_to_consume(queue_fog, AMQP_EXCHANGE_FOG,
-					AMQP_EVENT_DATA_UPDATE);
-	if (err) {
-		hal_log_error("Error on set up queue to consume.\n");
-		return -1;
-	}
-
-	err = amqp_set_queue_to_consume(queue_fog, AMQP_EXCHANGE_FOG,
-					AMQP_EVENT_DATA_REQUEST);
-	if (err) {
-		hal_log_error("Error on set up queue to consume.\n");
-		return -1;
-	}
-
-	err = amqp_set_queue_to_consume(queue_fog, AMQP_EXCHANGE_FOG,
-					   AMQP_EVENT_DEVICE_UNREGISTERED);
-	if (err) {
-		hal_log_error("Error on set up queue to consume.\n");
-		return -1;
+	for (i = 0; fog_events[i] != NULL; i++) {
+		err = amqp_set_queue_to_consume(queue_fog, AMQP_EXCHANGE_FOG,
+					fog_events[i]);
+		if (err) {
+			hal_log_error("Error on set up queue to consume.\n");
+			return -1;
```

Comment:
Shouldn't you call `amqp_bytes_free(queue_fog);` in case of failure?

Reply:
Done